internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixReadOnlySchemaRepository.java

```diff
@@ -372,23 +392,37 @@ public SchemaEntry getSupersetOrLatestValueSchema(String storeName) {
 
   @Override
   public Optional<SchemaEntry> getSupersetSchema(String storeName) {
-    schemaLock.readLock().lock();
+    getSchemaLock().readLock().lock();
     try {
       fetchStoreSchemaIfNotInCache(storeName);
-      SchemaData schemaData = schemaMap.get(getZkStoreName(storeName));
+      SchemaData schemaData = getSchemaMap().get(getZkStoreName(storeName));
       if (schemaData == null) {
         throw new VeniceNoStoreException(storeName);
       }
       Optional<Integer> supersetSchemaID = getSupersetSchemaID(storeName);
+      if (supersetSchemaID.isPresent()) {
+        if (schemaData.getValueSchema(supersetSchemaID.get()) == null) {
```

Comment:
Should we do a fore refresh if the derived schema for the superset schema doesn't exist?

Also for the force refresh, shall we do it multiple times (such as 3 times with exponential delay)?
It is not guaranteed the instant force-refresh will get the schemas since the server might not be connecting to the most-fresh ZK nodes.

Reply:
+1 to retry.

Reply:
Okay, regarding derived schema - maybe I can only check if it has at least one derived schema related if write compute is enabled for the store (cuz superset schema can also be for the read compute if I understand it correctly)

Reply:
And it should also check RMD schema as well for sanity.

Reply:
yes, for RMD schema as well.

Reply:
Yeah, update schema is for write-compute only and we can skip it for read-compute only stores.

Reply:
Just updated, please feel free to take a second look